ocaml-lsp-server.opam

```diff
@@ -23,6 +23,7 @@ depends: [
   "ocamlfind" {>= "1.5.2"}
   "ocaml" {>= "4.06"}
   "dune" {>= "2.0"}
+  "cmdliner"
```

Comment:
cmdliner is vendored here, so it's not an opam dependency
Also, note for the future, the `opam` file is generated automatically by dune. So if cmdliner was to be added as a dependency, it should be done by adding it in the `dune-package` file